src/azure-cli-core/tests/test_api_profiles.py

```diff
@@ -0,0 +1,126 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+import mock
+
+from azure.cli.core.profiles import (get_api_version,
+                                     supported_api_version,
+                                     ResourceType)
+from azure.cli.core.profiles._shared import APIVersionException
+from azure.cli.core.cloud import Cloud
+
+
+class TestAPIProfiles(unittest.TestCase):
+
+    @mock.patch('azure.cli.core._profile.CLOUD', Cloud('TestCloud', profile='myprofile'))
+    def test_get_api_version(self):
+        ''' Can get correct resource type API version '''
+        test_profile = {'myprofile': {ResourceType.MGMT_STORAGE: '2020-10-10'}}
+        with mock.patch('azure.cli.core.profiles._shared.AZURE_API_PROFILES', test_profile):
+            self.assertEqual(get_api_version(ResourceType.MGMT_STORAGE), '2020-10-10')
+
+    @mock.patch('azure.cli.core._profile.CLOUD', Cloud('TestCloud', profile='myprofile'))
+    def test_get_api_version_invalid_rt(self):
+        ''' Resource Type not in profile '''
+        test_profile = {'myprofile': {ResourceType.MGMT_STORAGE: '2020-10-10'}}
+        with mock.patch('azure.cli.core.profiles._shared.AZURE_API_PROFILES', test_profile):
+            with self.assertRaises(APIVersionException):
+                get_api_version(ResourceType.MGMT_COMPUTE)
+
+    @mock.patch('azure.cli.core._profile.CLOUD', Cloud('TestCloud', profile='not-a-real-profile'))
+    def test_get_api_version_invalid_active_profile(self):
+        ''' The active profile is not in our profile dict '''
+        test_profile = {'myprofile': {ResourceType.MGMT_STORAGE: '2020-10-10'}}
+        with mock.patch('azure.cli.core.profiles._shared.AZURE_API_PROFILES', test_profile):
+            with self.assertRaises(APIVersionException):
+                get_api_version(ResourceType.MGMT_STORAGE)
+
+    @mock.patch('azure.cli.core._profile.CLOUD', Cloud('TestCloud', profile='myprofile'))
+    def test_get_api_version_invalid_rt_2(self):
+        ''' None is not a valid resource type '''
+        test_profile = {'myprofile': {ResourceType.MGMT_STORAGE: '2020-10-10'}}
+        with mock.patch('azure.cli.core.profiles._shared.AZURE_API_PROFILES', test_profile):
+            with self.assertRaises(APIVersionException):
+                get_api_version(None)
+
+    def test_supported_api_profile_no_constraints(self):
+        self.assertTrue(supported_api_version(None))
```

Comment:
I assume it makes sure the default returned value from this function is not `None`. @derekbekoe a line of comments explaining the intention would be helpful.